### PR TITLE
refactor(behavior_path_planner): move turn_signal_on_swerving param to bpp.param.yaml

### PR DESCRIPTION
**Pull request by beyzanurkaya**
Signed-off-by: beyza <bnk@leodrive.ai>

## Description

I moved turn_signal_on_swerving param to bpp.param.yaml

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/behavior_path_planner.param.yaml [hunk 1]
```diff
@@ -18,4 +18,5 @@
     turn_signal_minimum_search_distance: 10.0
     turn_signal_search_time: 3.0
     turn_signal_shift_length_threshold: 0.3
+    turn_signal_on_swerving: false # turn signal on while swerving
```
Comment (purewater0901) [inline launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/behavior_path_planner.param.yaml:21]: default should be `true`

File: planning/behavior_path_planner/src/behavior_path_planner_node.cpp [hunk 2]
```diff
@@ -221,6 +221,8 @@ BehaviorPathPlannerParameters BehaviorPathPlannerNode::getCommonParam()
   p.turn_signal_search_time = declare_parameter("turn_signal_search_time", 3.0);
   p.turn_signal_shift_length_threshold =
     declare_parameter("turn_signal_shift_length_threshold", 0.3);
+  p.turn_signal_on_swerving = declare_parameter("turn_signal_on_swerving", false);
```
Comment (purewater0901) [inline planning/behavior_path_planner/src/behavior_path_planner_node.cpp:224]: default value should be `true`

**Comment by purewater0901** [comment]
@beyzanurkaya Thank you for your contribution. I just add some comments about your default value. Could you change them to `true`?

**Comment by purewater0901** [comment]
Please add parameter explanations [here](https://github.com/autowarefoundation/autoware.universe/blob/main/planning/behavior_path_planner/behavior_path_planner_turn_signal-design.md#parameters-for-turn-signal-decider)

**Comment by beyzanurkaya** [comment]
@purewater0901 I did the necessary changes.

**Comment by beyzanurkaya** [comment]
@purewater0901 It's ready for review.

**Review by purewater0901 — APPROVED** [review]
Perfect! Thank you for your contribution!

**Pull request merged**